opshin/__main__.py

```diff
@@ -63,6 +138,12 @@ def main():
         choices=Command.__members__.keys(),
         help="The command to execute on the input file.",
     )
+    a.add_argument(
+        "purpose",
+        type=str,
+        choices=Purpose.__members__.keys(),
+        help="The intended script purpose. Determines the number of parameters.",
+    )
```

Comment:
Should we make the default purpose spending or always force a choice? What is the current assumption of purpose? Can we display the number of parameters / which ones are necessary in the help text?

Reply:
The default purpose currently is "any". I think forcing a choice is necessary otherwise people will just default to not specifying anything and bump into this issue later.

good idea to display the number of parameters for each purpose in the help text!